File: VB/NorthwindDataProvider.vb
Imports System.Data
Imports System.Data.SqlClient

Namespace RichEditImageMailMerge

    Public Module NorthwindDataProvider

        Private connectionString As String = "Data Source=.\SQLExpress;Initial Catalog=Northwind;Integrated Security=SSPI"

        Sub New()
            Using connection As SqlConnection = New SqlConnection(connectionString)
                Dim selectCommand As SqlCommand = New SqlCommand("SELECT * FROM Categories", connection)
                Dim da As SqlDataAdapter = New SqlDataAdapter(selectCommand)
                categoriesField = New DataTable("Categories")
                da.Fill(categoriesField)
                categoriesField.Constraints.Add("IDPK", categoriesField.Columns("CategoryID"), True)
                selectCommand.Dispose()
            End Using
        End Sub

        Private categoriesField As DataTable

        Public ReadOnly Property Categories As DataTable
            Get
                Return categoriesField
            End Get
        End Property
    End Module
End Namespace

File: CS/NorthwindDataProvider.cs
using System.Data;
using System.Data.SqlClient;

namespace RichEditImageMailMerge {
    public static class NorthwindDataProvider {
        private static string connectionString = @"Data Source=.\SQLExpress;Initial Catalog=Northwind;Integrated Security=SSPI";

        static NorthwindDataProvider() {
            using (SqlConnection connection = new SqlConnection(connectionString)) {
                SqlCommand selectCommand = new SqlCommand("SELECT * FROM Categories", connection);
                SqlDataAdapter da = new SqlDataAdapter(selectCommand);

                categories = new DataTable("Categories");

                da.Fill(categories);

                categories.Constraints.Add("IDPK", categories.Columns["CategoryID"], true);
 
                selectCommand.Dispose();
            }
        }

        private static DataTable categories;

        public static DataTable Categories {
            get {
                return categories;
            }
        }
    }
}